### PR TITLE
Get Admin Pharmacies

### LINKED ISSUE #36: [GET] admin/pharmasies?status=pending

### PR DESCRIPTION
**Pull request by Ibtisam-Hemmo**

### CODE REVIEW

File: server/app.ts [hunk 1]
```diff
@@ -24,5 +24,6 @@ if (NODE_ENV === 'production') {
     res.sendFile(join(__dirname, '..', 'client', 'build', 'index.html'));
   });
 }
+app.use(adminRouter);
```
Comment (Raghad-Mezied) [inline server/app.ts]: Use all your routers in the index of the router folder, and then use it here

File: server/controllers/admin/allPharmacies.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { Request, Response, NextFunction } from "express";
+import {getAllPharmaciesGeneral} from '../../queries/admin';
+interface RequestQuery {
+  status: string,
+  page: number
+}
+
+const getAllPharmacies = async (req: Request<RequestQuery>, res:Response, next:NextFunction)=>{
+    const {status, page = 1 } = req.query;
```
Comment (Raghad-Mezied) [inline server/controllers/admin/allPharmacies.ts]: validate

File: server/queries/admin/allPharmacies.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { Pharmacy } from '../../models';
+
+const getAllPharmaciesGeneral = async (status: any, page: any) =>{
+  if(status){
+    return Pharmacy.findAndCountAll({
+      where: {
+        status: status,
+      },
+      offset: page,
```
Comment (Raghad-Mezied) [inline server/queries/admin/allPharmacies.ts]: Check what the offset does exactly so that you can give it the correct value.

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

**Review by Raghad-Mezied — DISMISSED** [review]

**Review by Nader-SH — DISMISSED** [review]

File: server/queries/admin/allPharmacies.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { Pharmacy } from '../../models';
+
+const getAllPharmaciesGeneral = async (status: any, page: any) =>{
```
Comment (AhmedAbuSharar) [inline server/queries/admin/allPharmacies.ts]: edit type for ( status, page )

**Review by AhmedAbuSharar — CHANGES_REQUESTED** [review]

File: server/queries/admin/allPharmacies.ts (hunk 4 above)
Comment (Ibtisam-Hemmo) [reply server/queries/admin/allPharmacies.ts]: We are getting status and page form query and it has many types and you can not restrict it to only having one type.
So both do the job whether being type any or assigning all the types that query you are getting from the request.
Plus there is schema to validate that status and page num we are getting from the request before sending it to the query to get all pharmacies

**Review by Raghad-Mezied — DISMISSED** [review]

**Review by Nader-SH — CHANGES_REQUESTED** [review]
fix the tests no one pass

**Review by Raghad-Mezied — DISMISSED** [review]

**Review by Raghad-Mezied — DISMISSED** [review]

**Pull request merged**